bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5938,6 +6043,205 @@ private static void closeStatement(Statement statement) {
 	}
 
 	/**
+	 * Sets the end date for the given ingest job
+	 *
+	 * @param ingestJobId The ingest job to set the end date for
+	 * @param endDate     The end date
+	 *
+	 * @throws TskCoreException If inserting into the database fails
+	 * @throws TskDataException If the ingest job is not found, or already has
+	 *                          an end time.
+	 */
+	void setIngestJobEndDateTime(int ingestJobId, long endDateTime) throws TskCoreException, TskDataException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		ResultSet resultSet = null;
+		try {
+			Statement statement = connection.createStatement();
+			resultSet = statement.executeQuery("SELECT ingest_job_id FROM ingest_jobs WHERE ingest_job_id=" + ingestJobId);
+			if (!resultSet.next()) {
+				statement.executeUpdate("UPDATE ingest_jobs SET end_date_time=" + endDateTime + " WHERE ingest_job_id=" + ingestJobId + ";");
+			} else {
+				throw new TskDataException("Given ingest job was not found in database.");
+			}
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error updating the end date.", ex);
+		} finally {
+			closeResultSet(resultSet);
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	void setIngestStatus(int ingestJobId, IngestJobStatusType status) throws TskCoreException, TskDataException {
```

Comment:
Please remove the SELECT to check for the record being present and the TskDataException and add documentation.